docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+
+Libvirt exposes a set of hypervisor and operation-specific metrics.
+Examples are the migration statistics. Those metrics are usually
+available only while the operation is in progress, or shortly after.
+Let's call this set `hypervisor metrics`
+
+Lastly, VMs often run a Guest Agent, a deamon which periodically
+collects and reports back to the hypervisor another set of metrics.
+Let's call this set `guest metrics`
+
+The last two sets have no counterpart in the containers ecosystem, but
+are meaningful for VM management. The following table provides a
+breakdown of the major VM metrics.
+
+| Metric                             | Group         | Ephemeral | Provided by |
+| ---------------------------------- |:-------------:|:---------:|:-----------:|
+| CPU time total                     | system        | -         | -           |
+| CPU time user                      | system        | -         | -           |
+| CPU time system                    | system        | -         | -           |
+| nic #x RX/TX bytes                 | system        | -         | -           |
+| nic #x RX/TX packets               | system        | -         | -           |
+| nic #x RX/TX errors                | system        | -         | -           |
+| nic #x RX/TX dropped packets       | system        | -         | -           |
+| block device #x R/W operations     | system        | -         | -           |
+| block device #x R/W bytes          | system        | -         | -           |
+| block device #x R/W total time     | system        | -         | -           |
+| block device #x flush operations   | system        | -         | -           |
+| block device #x flush total time   | system        | -         | -           |
+| block device #x allocation         | system        | -         | -           |
+| block device #x capacity           | system        | -         | -           |
+| block device #x physical           | system        | -         | -           |
+| block device #x threshold          | system        | -         | -           |
+| guest state                        | hypervisor    | -         | -           |
+| guest state reason                 | hypervisor    | -         | -           |
+| memory balloon current             | hypervisor    | -         | -           |
+| memory balloon maximum             | hypervisor    | -         | -           |
+| virtual CPU count                  | hypervisor    | -         | -           |
+| virtual CPU maximum                | hypervisor    | -         | -           |
+| virtual CPU #x state               | hypervisor    | -         | -           |
+| virtual CPU #x total time          | hypervisor    | -         | -           |
+| network interface count            | hypervisor    | -         | -           |
+| block device count                 | hypervisor    | -         | -           |
+| migration downtime                 | hypervisor    | yes       |             |
```

Comment:
Any reason for not continuing to use a `-` in The _provided by_ column?

Reply:
It was a placeholder. I was looking for the equivalent concept in kubernetes, initial research was unfruitful. Now I learned those are 'core metrics', so I can drop the column.